docs/source/format/Intro.rst

```diff
@@ -0,0 +1,454 @@
+.. figure:: ./images/dense-union-diagram.svg
+   :alt: Diagram is showing the difference between the dense union data type
+         presented in a Table and the dataactually stored in computer
+         memory.
+
+   Physical layout diagram for dense union data type.
+
+Sparse union
+^^^^^^^^^^^^
+
+A sparse union has the same structure as a dense union, with the omission of the offsets
+array. In this case, the child arrays are each equal in length to the length of the union.
+
+
+.. figure:: ./images/sparse-union-diagram.svg
+   :alt: Diagram is showing the difference between the sparse union data type
+         presented in a Table and the dataactually stored in computer
+         memory.
+
+   Physical layout diagram for sparse union data type.
+
+Dictionary Encoded Layout
+=========================
+
+Dictionary encoding can be effective when you have data with many repeated values.
+The values are represented by integers referencing a dictionary usually consisting of
+unique values.
+
+.. figure:: ./images/dictionary-diagram.svg
+   :alt: Diagram is showing the difference between the dictionary data type
+         presented in a Table and the dataactually stored in computer
+         memory.
+
+   Physical layout diagram for dictionary data type.
+
+Run-End Encoded Layout
+======================
+
+Run-end encoding is well-suited for representing data containing sequences of the
+same value. These sequences are called runs. Run-end encoded array has no buffers
+by itself, but has two child arrays:
+
+*  **Run ends array:** holds the index in the array where each run ends.
+*  **Values array:** the actual values without repetitions.
+
+.. figure:: ./images/ree-diagram.svg
+   :alt: Diagram is showing the difference between the run-end encoded data
+         type presented in a Table and the dataactually stored in computer
+         memory.
+
+   Physical layout diagram for run-end encoded data type.
+
+.. link to All types overview https://github.com/apache/arrow/issues/14752
+
+Overview of Arrow Terminology
+=============================
+
+**Physical layout**
+A specification for how to arrange values of an array in memory.
+
+**Buffer**
+A contiguous region of memory with a given length. Buffers are used to store data for arrays.
+
+**Array**
+A contiguous, one-dimensional sequence of values with known length where all values have the
+same type. An array consists of zero or more buffers.
+
+**Chunked Array**
+A discontiguous, one-dimensional sequence of values with known length where all values have
+the same type. Consists of zero or more arrays, the “chunks”.
+
+.. note::
+   Chunked Array is a concept specific to certain implementations such as Arrow C++ and PyArrow.
+
+**RecordBatch**
+A contiguous, two-dimensional data structure which consist of ordered collection of arrays
+of the same length.
+
+**Schema**
+A collection of fields with optional metadata that determines all the data types of an object
+like a RecordBatch or Table.
```

Comment:
Decided to add it under the schema, not as a specific "term", because it is very specific to the RecordBatch/Table (just like the schema though) and we use fields in this document that are connected to the types and their layouts (struct for example). Feel it might add some confusion. Could be wrong!

https://github.com/apache/arrow/pull/41593/commits/44aeebf45499c433f37f58365baee9756d617dad